pep-0707.rst

```diff
@@ -0,0 +1,358 @@
+PEP: 707
+Title: PEP 707: A simplified signature for __exit__ and __aexit__
+Author: Irit Katriel <iritkatriel@gmail.com>
+Discussions-To:
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 18-Feb-2023
+Python-Version: 3.12
+Post-History:
+Resolution:
+
+
+Abstract
+========
+
+This PEP proposes to make the intrpreter accept context managers whose
+``__exit__``/``__aexit__`` method takes only a single exception instance,
+while continuing to support also the current ``(typ, exc, tb)`` signature
+for backwards compatibility.
+
+This change would remove a redundancy in the language which is
+confusing for those learning and using it. In addition, it will remove
+some of the overhead of supporting context mamangers in the interpreter:
+micro-benchmarks showed a 13% speedup for entering and exiting a context
+manager with an empty body.
+
+The proposed implementation uses introspection, which is tailored to the
+requirements of this use case. The solution ensures the safety of the new
+feature by supporting it only in non-ambiguous cases. In particular, any
+signature that *could* accept three arguments, is assumed to expect them.
+
+
+Motivation
+==========
+
+In the past, an exception was represented in many parts of Python by a
+tuple of three elements: the type of the exception, its value, and its
+traceback.  While there were good reasons for this design at the time,
+they no longer hold because the type and traceback can now be reliably
+deduced from the exception instance. Over the last few years we saw
+several efforts to simplify the representation of exceptions.
+
+Since 3.10, the ``traceback`` module's functions accept either a 3-tuple
+as described above, or just an exception instance as a single argument
+[https://github.com/python/cpython/issues/70577].
+
+Internally, the interpreter no longer represents exceptions as a triplet.
+This was removed for the handled exception in 3.11 
+[https://github.com/python/cpython/pull/30122] and
+for the raised exception in 3.12
+[https://github.com/python/cpython/pull/101607]. As a consequence,
+several APIs that expose the triplet were deprecated in favour of
+simpler alternatives:
+
+list-table:: Exception Related APIs
```

Comment:
This table isn't rendering, see the preview at https://pep-previews--3018.org.readthedocs.build/pep-0707/#motivation